core/commands/partials/registration_manager.rb

```diff
@@ -70,7 +72,18 @@ def withdraw_subscription(filename)
 
   # Run request to deregister the system with the given credentials
   def run_deregistration_command(credentials)
-    command = "curl -k -X DELETE -u #{credentials['username']}:#{credentials['password']} #{URI.join(@registration_proxy_url, REGISTRATION_ENDPOINT)}"
+    registration_server_address = @use_proxy ? @registration_proxy_url : CUSTOMER_CENTER_URL
+    command = "curl -k -X DELETE -u #{credentials['username']}:#{credentials['password']} #{URI.join(registration_server_address, REGISTRATION_ENDPOINT)}"
     ShellCommands.run_command(@logger, command)
   end
+
+  # Check if the system can be registered via the Registration Proxy Server
+  def use_proxy?(machines_provider)
+    machines_provider == 'gcp' && proxy_available?
+  end
+
+  def proxy_available?
```

Comment:
This check can not be performed on the host machine with MDBCI. This check is viable only when run from the virtual machine in the cloud. You must copy the results of the proxy check from the VM instead.